### PR TITLE
Lg 303 codestyle

### LINKED ISSUE #303: adapt codestyle using black isort flake8
As mentioned in #286, `black`, `isort` and `flake8` formatted code could be implemented quiet easy.

- [ ] code formatting
- [ ] ci-tests

I think it makes sense to do these tests on the whole repository. So it will be mandatory to use the mentioned tools as soon as this issue is resolved.

### PR DESCRIPTION
**Pull request by Lukas113**
closes #303 

With this PR, the entire repository has been reformatted to pass **flake8**, **black** and **isort** checks. In addition I added documentation for installing and using these tools (in `doc.src.development.md`). GitHub-CI now has tests to check the entire repository. So if a developer does not adhere to a format we test against, it is simply not merged and thus everyone is forced to use these tools.

### CODE REVIEW

File: karabo/sourcedetection/evaluation.py [hunk 1]
```diff
@@ -351,11 +369,11 @@ def plot_flux_ratio_to_distance(
...
         )
         ra_dec_pred = truth[[0, 1]]
-        ra_ref = truth[0]
-        dec_ref = truth[1]
+        # ra_ref = truth[0]
+        # dec_ref = truth[1]
         flux_ref = truth[2]
-        ra_pred = pred[0]
-        dec_pred = pred[1]
+        # ra_pred = pred[0]
```
Comment (kenfus) [inline karabo/sourcedetection/evaluation.py:375]: remove

**Review by kenfus — CHANGES_REQUESTED** [review]

File: doc/src/conf.py [hunk 2]
```diff
@@ -12,7 +12,6 @@
 #
 import os
 import sys
-import sphinx_rtd_theme
```
[the remark this answers is not in the document]
Comment (Lukas113) [reply doc/src/conf.py:15]: I don't know much about sphinx. But *flake8* marked it as unused and I've never seen (apart from docstringtests) that a string needs an import.

File: karabo/sourcedetection/evaluation.py (hunk 1 above)
Comment (Lukas113) [reply karabo/sourcedetection/evaluation.py]: done

File: karabo/simulation/visibility.py [hunk 3]
```diff
@@ -91,7 +93,7 @@ def combine_spectral_foreground_vis(
         print("### Writing combined visibilities in ", combined_vis_filepath)
         start_row = 0
         exposure_sec = fg_header.get_time_average_sec()
-        sc = [0] * len(foreground_cross_correlation)
+        #  sc = [0] * len(foreground_cross_correlation)
```
[the remark this answers is not in the document]
Comment (Lukas113) [reply karabo/simulation/visibility.py]: done

File: doc/src/conf.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Lukas113) [reply doc/src/conf.py:15]: I've re-added it.

File: karabo/sourcedetection/result.py [hunk 4]
```diff
@@ -206,8 +212,11 @@ def __transform_bdsf_to_reduced_result_array(
...
-            wmsg = f"Got unexpected shape of `bdsf_detected_sources` of {bdsf_detected_sources.shape}, \
-                expected 2-dimensional array with sources!"
+            wmsg = (
+                "Got unexpected shape of `bdsf_detected_sources` of "
+                + f"{bdsf_detected_sources.shape}, expected 2-dimensional "
+                + "array with sources!"
+            )
             warn(KaraboWarning(wmsg))
             sources = bdsf_detected_sources
         return sources
```
[the remark this answers is not in the document]
Comment (Lukas113) [reply karabo/sourcedetection/result.py:222]: What do you mean by that? I don't see any files below.

[the remark this answers is not in the document]
Comment (Lukas113) [reply karabo/sourcedetection/result.py:222]: Yeah you're right, the files `karabo/test/data/cst_like_beam_port_1.txt` and `karabo/test/data/cst_like_beam_port_2.txt` were somehow changed.

Unfortunately, I was not able to reproduce this error. When I called one of the tools for the whole repository or for these specific files, they were not changed at all. So I think we need to be careful about which files we commit. If this error occurs again in the future, I will take a closer look.

**Review by kenfus — APPROVED** [review]

**Pull request merged**